basiclibrary/src/test/java/basiclibrary/LibraryTest.java

```diff
@@ -63,4 +64,38 @@
 
         assertArrayEquals(returnValue, Library.arrayWithTheLowestAverage(arrayOfTestData));
     }
+
+    @Test public void testAnalyzingWeatherDataMethod() {
+        int[][] testArr = {
+                {66, 64, 58, 65, 71, 57, 60},
+                {57, 65, 65, 70, 72, 65, 51},
+                {55, 54, 60, 53, 59, 57, 61},
+                {65, 56, 55, 52, 55, 62, 57}
+        };
+
+        assertEquals("High: 72\n" +
+                "Low: 51\n" +
+                "Never saw temperature: 63\n" +
+                "Never saw temperature: 67\n" +
+                "Never saw temperature: 68\n" +
+                "Never saw temperature: 69\n", Library.analyzingWeatherData(testArr));
```

Comment:
Challenge yourself to build the test to get the desired output without the extra return on the last line.